lib/components/viewers/stop-viewer.js

```diff
@@ -423,19 +424,30 @@ function getFormattedStopTime (stopTime, homeTimezone, soonText = 'Due', timeFor
 
   const now = moment().tz(homeTimezone)
   const serviceDay = moment(stopTime.serviceDay * 1000).tz(homeTimezone)
+  // Determine if arrival occurs on different day, making sure to account for
+  // any extra days added to the service day if it arrives after midnight. Note:
+  // this can handle the rare (and non-existent?) case where an arrival occurs
+  // 48:00 hours (or more) from the start of the service day.
+  const departureTimeRemainder = stopTime.realtimeDeparture % ONE_DAY_IN_SECONDS
+  const daysAfterServiceDay = (stopTime.realtimeDeparture - departureTimeRemainder) / ONE_DAY_IN_SECONDS
+  const arrivalDay = serviceDay.add(daysAfterServiceDay, 'day')
```

Comment:
To be consistent, this should be renamed `departureDay` since the data is derived from the departure time.

Reply:
I'm also going to refactor the `differentDay` boolean to be `vehicleDepartsToday`, which I think is a little more clear and make it less likely to be confused with `serviceDay` and `departureDay`.